src/tasks/minions/bso/memoryHarvestActivity.ts

```diff
@@ -0,0 +1,248 @@
+					energyAmount = increaseNumByPercent(energyAmount, 3);
+				}
+				loot.add(energy.item, Math.ceil(energyAmount));
+				break;
+			}
+			case MemoryHarvestType.ConvertWithEnergyToXP: {
+				cost.add(energy.item, Math.ceil(energyPerMemory * memoriesHarvested));
+				break;
+			}
+		}
+
+		let clueChance = 1200;
+		// Step 3: Roll for pet
+		for (let t = 0; t < memoriesHarvested; t++) {
+			if (roll(petChance)) {
+				loot.add('Doopy');
+			}
+			if (hasDivineHand && roll(clueChance) && 'clueTable' in energy && energy.clueTable) {
+				loot.add(energy.clueTable.roll());
+			}
+		}
+	}
+
+	if (hasGuthixianBoost) {
+		totalDivinationXP = increaseNumByPercent(totalDivinationXP, 20);
+	}
+
+	if (hasWispBuster) {
+		totalDivinationXP = increaseNumByPercent(totalDivinationXP, inventionBoosts.wispBuster.xpIncreasePercent);
+		boosts.push(`${inventionBoosts.wispBuster.xpIncreasePercent}% extra XP for Wisp-buster`);
+	}
+
+	if (hasBoon) {
+		boosts.push('10% extra XP for Boon');
+	}
+
+	if (hasGuthixianBoost) {
+		boosts.push('20% extra XP and energy for Guthixian Cache boost');
+	}
+
+	if (hasDivineHand) {
+		boosts.push(`${inventionBoosts.divineHand.memoryHarvestExtraYieldPercent}% extra energy for Divine hand`);
+	}
+
+	if (isUsingDivinationPotion) {
+		totalDivinationXP = increaseNumByPercent(totalDivinationXP, 3);
+		boosts.push('3% extra XP/energy for Divination potion');
+	}
+
+	return {
+		cost,
+		loot,
+		totalDivinationXP,
+		totalMemoriesHarvested,
+		petChancePerMemory: petChance,
+		avgPetTime: calculateAverageTimeForSuccess((totalMemoriesHarvested / petChance) * 100, duration),
+		boosts,
+		energyPerMemory
+	};
+}
+
+export const memoryHarvestTask: MinionTask = {
+	type: 'MemoryHarvest',
+	async run(data: MemoryHarvestOptions) {
+		let {
+			userID,
+			channelID,
+			duration,
+			e: energyItemID,
+			t: harvestMethodIndex,
+			wb: hasWispBuster,
+			dh: hasDivineHand,
+			dp: isUsingDivinationPotion
+		} = data;
+		const user = await mUserFetch(userID);
+		const energy = divinationEnergies.find(t => t.item.id === energyItemID)!;
+		const hasBoon = energy.boonBitfield !== null ? user.bitfield.includes(energy.boonBitfield) : false;
+		let didGetGuthixianBoost = false;
+		if (user.user.guthixian_cache_boosts_available > 0) {
+			await user.update({
+				guthixian_cache_boosts_available: {
+					decrement: 1
+				}
+			});
+			didGetGuthixianBoost = true;
+		}
+
+		const { boosts, totalDivinationXP, totalMemoriesHarvested, petChancePerMemory, loot, avgPetTime, cost } =
+			memoryHarvestResult({
+				duration,
+				hasBoon,
+				energy,
+				harvestMethod: harvestMethodIndex,
+				hasWispBuster,
+				hasGuthixianBoost: didGetGuthixianBoost,
+				hasDivineHand,
+				isUsingDivinationPotion
+			});
+
+		if (cost.length > 0) {
+			if (!user.owns(cost)) {
+				return handleTripFinish(
+					user,
+					channelID,
+					`${user}, ${user.minionName} couldn't complete the trip because they didn't have the required items: ${cost}.`,
```

Comment:
I'm not a fan of trips failing because you don't have the required items.

A lot of time wasted.

Plus, virtually every other task removes the required items in advance, so people will not expect they can't use their energy to make portents/etc...

Reply:
It only happens if they lose the required items mid-trip which shouldn't be something happening anyway, maybe once accidentally.